### PR TITLE
upgrade mybatis-plus to 3.4.3

### LINKED ISSUE #8513: [Upgrade] Upgrade mybatis-plus to 3.4.3
### Discussed in https://github.com/apache/dolphinscheduler/discussions/8162

<div type='discussions-op-text'>

<sup>Originally posted by **YiuTerran** January 24, 2022</sup>
Mybatis-plus v3.2.0 using mybatis 3.5.2. 

There are high level security problems for this version. See [CVE-2020-26945](https://nvd.nist.gov/vuln/detail/CVE-2020-26945)</div>

### PR DESCRIPTION
**Pull request by lenboo**
## Purpose of the pull request

fix #8513  upgrade mybatis-plus from 3.2.0 to 3.4.3

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Review by zhongjiajie — DISMISSED** [review]
License LGTM, do you have time to make a double-check? @CalvinKirs

**Review by CalvinKirs — DISMISSED** [review]
I have confirmed, No license changes.

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/datasource/SpringConnectionFactory.java [hunk 1]
```diff
@@ -58,12 +56,16 @@ public SqlSessionFactory sqlSessionFactory(DataSource dataSource) throws Excepti
         configuration.setCacheEnabled(false);
         configuration.setCallSettersOnNulls(true);
         configuration.setJdbcTypeForNull(JdbcType.NULL);
-        configuration.addInterceptor(paginationInterceptor());
+        configuration.setObjectWrapperFactory(new MybatisMapWrapperFactory());
+
+        MybatisPlusInterceptor mybatisPlusInterceptor = new MybatisPlusInterceptor();
+        mybatisPlusInterceptor.addInnerInterceptor(new PaginationInnerInterceptor());
+        mybatisPlusInterceptor.addInnerInterceptor(new OptimisticLockerInnerInterceptor());
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/datasource/SpringConnectionFactory.java]: Does it need to add this optimistic lock plugin? This plugin requires the object to have a @version field to take effect.

Comment (lenboo) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/datasource/SpringConnectionFactory.java]: You are right, i will remove this.

**Review by CalvinKirs — APPROVED** [review]
Well done

**Pull request merged**